### ISSUE TITLE
"Imgur upload failed (417) Internal expectation failed." When uploading to Imgur.

### ISSUE THREAD

**Issue by Zejkie** [ShareX/ShareX#4941]
When capturing a region or trying to upload an image from program directly, I get "Imgur upload failed (417) Internal expectation failed."
I have tried restarting, re-installing, and reauthorizing to Imgur to no success. 

Here's the error log with log file attached below:
"Imgur upload failed: (417) Internal expectation failed

Error message:
The remote server returned an error: (417) Expectation Failed.

Request URL:
https://api.imgur.com/3/upload

Status code:
(417) Expectation Failed

Headers:
Connection: close
X-Post-Rate-Limit-Limit: 1250
X-Post-Rate-Limit-Remaining: 1250
X-Post-Rate-Limit-Reset: 3600
X-Ratelimit-Clientlimit: 12500
X-Ratelimit-Clientremaining: 12500
X-Ratelimit-Clientreset: 86400
X-Ratelimit-Userlimit: 2000
X-Ratelimit-Userremaining: 2000
X-Ratelimit-Userreset: 3600
Content-Length: 116
Content-Type: application/json; charset=utf-8
Date: Wed, 22 Jul 2020 09:31:47 GMT

Response text:
{"status":417,"success":false,"data":{"error":"Internal expectation failed","request":"/3/upload","method":"POST"}}

Stack trace:
   at System.Net.HttpWebRequest.GetResponse()
   at ShareX.UploadersLib.Uploader.SendRequestFile(String url, Stream data, String fileName, String fileFormName, Dictionary`2 args, NameValueCollection headers, CookieCollection cookies, HttpMethod method, String contentType, String relatedData)"

[ShareX-Log-2020-07.txt](https://github.com/ShareX/ShareX/files/4959223/ShareX-Log-2020-07.txt)

**Comment by Jaex** [comment]
Imgur gives that error because you are trying to upload bmp file.

**Comment by rhemz** [comment]
I also get this same error when attempting to upload any image or recorded gif.  Error is the same when uploading anonymously or while logged in.

```
Imgur upload failed: (417) Internal expectation failed

Error message:
The remote server returned an error: (417) Expectation Failed.

Request URL:
https://api.imgur.com/3/upload

Status code:
(417) Expectation Failed

Headers:
Connection: close
X-Post-Rate-Limit-Limit: 1250
X-Post-Rate-Limit-Remaining: 1250
X-Post-Rate-Limit-Reset: 3600
X-Ratelimit-Clientlimit: 12500
X-Ratelimit-Clientremaining: 12500
X-Ratelimit-Clientreset: 86400
X-Ratelimit-Userlimit: 2000
X-Ratelimit-Userremaining: 2000
X-Ratelimit-Userreset: 3600
Content-Length: 116
Content-Type: application/json; charset=utf-8
Date: Tue, 04 Aug 2020 16:00:10 GMT

Response text:
{"status":417,"success":false,"data":{"error":"Internal expectation failed","request":"/3/upload","method":"POST"}}


Stack trace:
   at System.Net.HttpWebRequest.GetResponse()
   at ShareX.UploadersLib.Uploader.SendRequestFile(String url, Stream data, String fileName, String fileFormName, Dictionary`2 args, NameValueCollection headers, CookieCollection cookies, HttpMethod method, String contentType, String relatedData)
```

**Comment by louhow** [comment]
I don't think this is related to ShareX.

I saw the same thing as you when doing this:

```
curl -sH "Authorization: Client-ID MY_ID" -F "image=@out.gif" "https://api.imgur.com/3/upload"
```

But checking against this endpoint:

```
curl -s -H "Authorization: Client-ID MY_ID" -H "Expect: " -F "image=@out.gif" https://api.imgur.com/3/image.xml
```

I got a more helpful error saying that the file size was too large. I'm unsure why, as my gif was well below the limit laid out here in the imgur docs - https://help.imgur.com/hc/en-us/articles/115000083326-What-files-can-I-upload-What-is-the-size-limit-#:~:text=The%20maximum%20file%20size%20for%20animated%20images%20(like%20GIFs)%20and,holders%20will%20be%20lossily%20compressed.

Whatever the reason, decreasing the file size fixed my issue.
